recipe/schemas.py

```diff
@@ -266,7 +277,282 @@ def _coerce_filter(v):
+    }
+)
+
+
+class ConditionPost(object):
+    """ Convert an operator like 'gt', 'lt' into '_op' and '_op_value'
+    for easier parsing into SQLAlchemy """
+
+    def __init__(self, operator, _op):
+        self.operator = operator
+        self._op = _op
+
+    def __call__(self, value):
+        value['_op'] = self._op
+        value['_op_value'] = value.get(self.operator)
+        value.pop(self.operator)
+        return value
+
+
+def _condition_schema(operator, _op, scalar=True, use_registry=False):
+    if scalar:
+        allowed_values = [S.Integer(), S.String(), S.Float(), S.Boolean()]
+    else:
+        allowed_values = [S.List()]
+
+    if use_registry:
+        field = 'non_aggregated_field'
+    else:
+        field = non_aggregated_field_schema
+
+    _condition_schema = S.Dict(
+        allow_unknown=False,
+        schema={'field': field,
+                operator: {
+                    'anyof': allowed_values
+                }},
+        coerce_post=ConditionPost(operator, _op)
+    )
+    return _condition_schema
+
+
+def _full_condition_schema(use_registry=False):
+    """ Conditions can be a field with an operator, like this yaml example
+
+    condition:
+        field: foo
+        gt: 22
+
+    Or conditions can be a list of and-ed and or-ed conditions
+
+    condition:
+        or:
+            - field: foo
+              gt: 22
+            - field: foo
+              lt: 0
+    """
+
+    # Handle conditions where there's an operator
+    operator_condition = S.DictWhenKeyExists({
+        'gt':
+            _condition_schema('gt', '__gt__', use_registry=use_registry),
+        'gte':
+            _condition_schema('gte', '__ge__', use_registry=use_registry),
+        'ge':
+            _condition_schema('ge', '__ge__', use_registry=use_registry),
+        'lt':
+            _condition_schema('lt', '__lt__', use_registry=use_registry),
+        'lte':
+            _condition_schema('lte', '__le__', use_registry=use_registry),
+        'le':
+            _condition_schema('le', '__le__', use_registry=use_registry),
+        'eq':
+            _condition_schema('eq', '__eq__', use_registry=use_registry),
+        'ne':
+            _condition_schema('ne', '__ne__', use_registry=use_registry),
+        'in':
+            _condition_schema(
+                'in', 'in_', scalar=False, use_registry=use_registry
+            ),
+        'notin':
+            _condition_schema(
+                'notin', 'notin', scalar=False, use_registry=use_registry
+            ),
+        'or': S.Dict(schema={'or': S.List(schema='operator_condition')}),
+        'and': S.Dict(schema={'and': S.List(schema='operator_condition')}),
+    },
+                                             required=False)
+
+    return {
+        'registry': {'operator_condition': operator_condition},
+        'schema_ref': 'operator_condition',
+    }
+
+
+condition_schema = _full_condition_schema(use_registry=False)
```

Comment:
AFAICT, this definition is only used in unit tests, and it's also the only thing that uses the `use_registry=False` parameter. I think this introduces a lot of unnecessary complexity. I think `_condition_schema` should unconditionally use the registry to refer to the `non_aggregated_field` schema, and we just need to make sure that `non_aggregated_field` is always available in a registry when validating a schema that uses a condition. This will get rid of a lot of wordiness in how we compose these schemas.

Reply:
:+1: will take a look